node/src/main/scala/coop/rchain/node/BlockStoreMigrate.scala

```diff
@@ -0,0 +1,66 @@
+package coop.rchain.node
+
+import coop.rchain.blockstorage.{FileLMDBIndexBlockStore, KeyValueBlockStore}
+import coop.rchain.casper.storage.RNodeKeyValueStoreManager
+import coop.rchain.metrics.Metrics
+import coop.rchain.rspace.Context
+import coop.rchain.shared.Log
+import coop.rchain.shared.syntax.sharedSyntaxKeyValueStoreManager
+import monix.eval.Task
+import org.rogach.scallop.ScallopConf
+import monix.execution.Scheduler.Implicits.global
+
+import java.nio.file.Path
+
+final case class Options(arguments: Seq[String]) extends ScallopConf(arguments) {
+  val width = 120
+  helpWidth(width)
+  printedName = "migrate-blockstore"
+
+  val originalBlockStore = opt[Path](
+    descr = "original block store directory",
+    required = true
+  )
+
+  val targetDataDir = opt[Path](
+    descr = "Output block store directory",
+    required = true
+  )
+
+  verify()
+
+}
+object BlockStoreMigrate {
+  def main(args: Array[String]): Unit = {
+    val options            = Options(args)
+    val originalBlockStore = options.originalBlockStore()
+    val targetDataDir      = options.targetDataDir()
+    implicit val log       = effects.log
+    implicit val metrics   = new Metrics.MetricsNOP[Task]
+
+    val t = for {
+      casperStoreManager <- RNodeKeyValueStoreManager[Task](targetDataDir)
+      blockKVStore       <- KeyValueBlockStore[Task](casperStoreManager)
+
+      blockFileStorage <- {
+        val blockstoreEnv = Context.env(originalBlockStore, 1024L * 1024L * 1024L)
+        for {
+          blockStore <- FileLMDBIndexBlockStore
+                         .create[Task](blockstoreEnv, originalBlockStore)
+                         .map(_.right.get)
+        } yield blockStore
+      }
+      _ <- log.info("Migrating file based block store to key value store.")
+
+      dataStream    <- blockFileStorage.iterateStream()
```

Comment:
Here `iterateStream` is called on FileBS instance so there is no need to have it on a BS trait.